pkg/skaffold/build/cluster/cluster.go

```diff
@@ -42,7 +46,46 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, tags tag.ImageTags,
 		defer teardownDockerConfigSecret()
 	}
 
-	return build.InParallel(ctx, out, tags, artifacts, b.buildArtifactWithKaniko)
+	var kanikoArtifacts []*latest.Artifact
+	var otherArtifacts []*latest.Artifact
+
+	for _, a := range artifacts {
+		if a.ArtifactType.KanikoArtifact != nil {
+			kanikoArtifacts = append(kanikoArtifacts, a)
+			continue
+		}
+		otherArtifacts = append(otherArtifacts, a)
+	}
+
+	pArts, err := build.InParallel(ctx, out, tags, kanikoArtifacts, b.buildArtifactWithKaniko)
+	if err != nil {
+		return nil, errors.Wrap(err, "building artifacts in parallel with kaniko")
+	}
+
+	sArts, err := build.InSequence(ctx, out, tags, otherArtifacts, b.runBuildForArtifact)
```

Comment:
I think we should run build.InSequence even if build.InParallel failed.  In general we want to try building all artifacts and hopefully with @nkubala eventing change, users will get notifications of build passed or failed as they occur.

In another PR, we are thinking of adding a `--fail-fast` flag in cases where users do not want builds to continue building if there is a failure. The way you have it right now is `--fail-fast` mode.

Reply:
sounds good, I changed it to run everything and return the error at the end